store/views.py

```diff
@@ -77,6 +91,40 @@ def loanBookView(request):
 @csrf_exempt
 @login_required
 def returnBookView(request):
-    pass
-
+    response_data = {
+        'message': 'failure',
+    }
+    # START YOUR CODE HERE
+    book_id = int(request.POST['book_id']) # get the book id from post data
+    bookCopy=get_object_or_404(BookCopy,id=book_id)
+    response_data['message']='success'
+    print(type(bookCopy.borrow_date))
+    BookCopy.objects.filter(id=book_id).update(borrower=None,status=True)
+    return JsonResponse(response_data)
 
+@csrf_exempt
+@login_required
+def rateBookView(request):
+    response_data = {
+        'message': None,
+    }
+    try:
+        book_id = int(request.POST.get('book_id')) # get the book id from post data
+        book=Book.objects.filter(id=book_id)
+        rating = int(request.POST.get('val'))
+        bookPrevRev=bookRating.objects.filter(bookName=book[0]).filter(user=request.user)
+        
+        if(len(bookPrevRev)==0):
+            bookRating.objects.create(bookName=book[0],user=request.user,ratingValue=rating)
+        else :
+            bookPrevRev.update(ratingValue=rating)
+        response_data['message']='success'
+        value=0
+        bookallRatings=bookRating.objects.filter(bookName=book[0])
+        for i in bookallRatings:
+            value +=  i.ratingValue
+        value=value/len(bookallRatings)
```

Comment:
This is correct but a more Django-ish way of doing it could be, using aggregate functions provided in Django ORM.
```python
bookRating.objects.filter(book__pk=book_id).aggregate(Avg('rating'))['rating__avg']
```